generators/app/index.js

```diff
@@ -178,70 +234,113 @@ module.exports = class extends Generator {
+    let subGenerators = env
+      .lookup({ localOnly: true, packagePaths: generatorPath })
+      .filter((sub) => {
+        const subGenerator = env.get(sub.namespace);
+        return !subGenerator.hidden;
+      });
 
+    // list the available subgenerators in the console (as help)
     if (this.options.list) {
-      const subgen = env
-        .lookup({ localOnly: true, packagePaths: generatorPath })
-        .filter((sub) => {
-          return !env.get(sub.namespace)?.hidden;
+      let maxLength = 0;
+      this.log(subGenerators
+        .map(sub => {
+          maxLength = Math.max(sub.namespace.length, maxLength);
+          return sub;
         })
-        .map((sub) => {
-          let line = `${deriveSubcommand(sub.namespace)}\t\t`;
-          if (deriveSubcommand(sub.namespace).length < 8) {
-            line += "\t";
+        .reduce((output, sub) => {
+          const subGenerator = env.get(sub.namespace);
+          const displayName = subGenerator.displayName || "";
+          let line = `  ${deriveSubcommand(sub.namespace).padEnd(maxLength + 2)}`;
+          if (displayName) {
+            line += ` # ${subGenerator.displayName}`;
           }
-          line += `${env.get(sub.namespace)?.displayName}`;
-          return line;
-        });
-      this.log(
-        `This generator offers ${subgen.length} options:\n\n${subgen.join(
-          "\n"
-        )}`
-      );
+          return `${output}\n${line}`;
+        }, `Subcommands (${subGenerators.length}):`));
       return;
     }
 
-    let subGenerators = env
-      .lookup({ localOnly: true, packagePaths: generatorPath })
-      .filter((sub) => {
-        if (this.options.subGenerator) {
-          return (
-            !env.get(sub.namespace)?.hidden &&
-            sub.namespace.includes(`:${this.options.subGenerator}`)
-          );
-        }
-        return !env.get(sub.namespace)?.hidden;
+    // if a subcommand is provided as argument, identify the matching subgenerator
+    // and remove the rest of the subgenerators from the list for later steps
+    if (this.options.subcommand) {
+      const selectedSubGenerator = subGenerators
+        .filter((sub) => {
+          // identify the subgenerator by subcommand
+          return new RegExp(`:${this.options.subcommand}$`).test(sub.namespace);
+        });
+      if (selectedSubGenerator.length == 1) {
+        subGenerators = selectedSubGenerator;
+      } else {
+        this.log(
+          `The generator ${chalk.red(
+            this.options.generator
+          )} has no subcommand ${chalk.red(
+            this.options.subcommand
+          )}. Please select an existing subcommand!`
+        );
+      }
+    }
+
+    // transform the list of the subgenerators and identify the 
+    // default subgenerator for the default selection
+    let defaultSubGenerator;
+    let maxLength = 0;
+    subGenerators = subGenerators
+      .map(sub => {
+        const generator = env.get(sub.namespace);
```

Comment:
prepare the required props and calculate the maxLength of the row to finally use padEnd to ensure a harmonized alignment